src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -227,6 +238,9 @@ function setInitialDOMProperties(
       // Noop
     } else if (registrationNameModules.hasOwnProperty(propKey)) {
       if (nextProp) {
+        if (__DEV__ && typeof nextProp !== 'function') {
```

Comment:
But I don't feel strongly. Didn't think about the object case where it could change.